### ISSUE TITLE
SeriLog Wrapper does not work without enabled application insights

### ISSUE THREAD

**Issue by aurinko-vicky** [serilog/serilog#1281]
I created a custom wrapper on SeriLog for .net core applications, which write logs to database. When I create a nuget package from my solution and try to use it in another projects, it does not log anything without enabled application insights. 
No exception or error appears. 
Does serilog has any dependency on application insights? 
Is that normal behavior that it does nothing without adding application insights. 
Any help on that would be helpful.

**Comment by aurinko-vicky** [comment]
Yes, Application Insights sink is not used. 

Some details: 
1. class library with logger wrapper is created; 
2. this project depends on Microsoft.AspNetCore.Http, Serilog.Enrichers.Environment, and System.Data.SqlClient;
3. created custom HttpContextEnricher (extends ILogEventEnricher), which depends on IHttpContextAccessor; 
4. created stored procedure sink(extends ILogEventSink), which triggers stored procedure to write logs to database, uses System.Data.SqlClient; 
5. logger is configured with extension method 

            loggerConfiguration.MinimumLevel.Is(logConfig.MinLogLevel)
                .Enrich.WithProperty("applicationName", appname)
                .Enrich.WithProperty("logger", loggerName)
                .Enrich.WithMachineName()
                .Enrich.With(new HttpContextEnricher(httpContextAccessor))
                .WriteTo.Sink(new StoredProcedureSink(connectionString));

If calling application does not have .UseApplicationInsights() in startup, the logger logs nothing.

**Comment by nblumhardt** [comment]
Sounds very odd! :-) Have you tried:

 * Writing an event directly through the Serilog logger, as soon as it's built, and
 * Capturing the output from Serilog's `SelfLog`? (See "Debugging and Diagnostics" on the wiki)

Best regards,
Nick

**Comment by aurinko-vicky** [comment]
I spend some more time on debugging and finally figured out what was the root of the problem. 
I saw that without .UseApplicationInsights(), HttpContext of IHttpContextAccessor was null, so in my HttpContextEnricher several properties have not been added to log event properties. Then lacking properties were not passed to stored procedure, exception has been raised and nothing happened. 

After I registered IHttpContextAccessor in calling app in Startup.cs, HttpContext got proper value and issue disappeared. 

In the app, which was used for testing logger wrapper, ApplicationInsights were enabled. Probably, they register IHttpContextAccessor under the hood and HttpContext had been got populated. 

That was just my misconception with proper registration of HttpContextAccessor, so now everything works fine. 

Anyway, thank you for quick responses!
